apps/lend/src/components/ChartOhlcWrapper/LiquidityData.tsx

```diff
@@ -0,0 +1,191 @@
+import type { LlammaControllerEvent } from '@/ui/Chart/types'
+import { LendingMarketTokens } from './types'
+
+import styled from 'styled-components'
+import { t } from '@lingui/macro'
+
+import networks from '@/networks'
+import { formatNumber, getFractionDigitsOptions } from '@/ui/utils'
+import { convertFullTime, convertTime, convertTimeAgo } from '@/components/ChartOhlcWrapper/utils'
+
+import Box from '@/ui/Box'
+import TokenIcon from '@/components/TokenIcon'
+import { Chip } from '@/ui/Typography'
+import Tooltip from '@/ui/Tooltip'
+
+type Props = {
+  lendControllerData: LlammaControllerEvent[]
+  chainId: ChainId
+  coins: LendingMarketTokens
+}
+
+const LiquidityData = ({ lendControllerData, chainId, coins }: Props) => {
+  const imageBaseUrl = networks[chainId].imageBaseUrl
+
+  return (
+    <>
+      {coins &&
+        lendControllerData.map((transaction, index) => {
+          return (
```

Comment:
Maybe put this in a separate `const renderTransactionRow = useCallback(...)` function to avoid creating such nested code.